<a href="https://colab.research.google.com/github/Grinbaum/novo/blob/main/C%C3%B3pia_de_An%C3%A1lisedados.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [17]:
import pandas as pd
import csv
dados = pd.read_csv('microdados.csv')
dados.head()


ano  edicao cidade_sede  ...    esporte                        evento medalha
0  1896  Summer      Athina  ...  Athletics  Athletics Men's 1,500 metres    Gold
1  1896  Summer      Athina  ...  Athletics    Athletics Men's 800 metres    Gold
2  1896  Summer      Athina  ...  Athletics      Athletics Men's Marathon     NaN
3  1896  Summer      Athina  ...     Tennis          Tennis Men's Doubles  Bronze
4  1896  Summer      Athina  ...     Tennis          Tennis Men's Singles     NaN

[5 rows x 15 columns]

In [ ]:
# Usamos as bibliotecas CSV e Pandas para abrir os dados

In [18]:
print(dados.describe()) 

                 ano      id_atleta  ...         altura           peso
count  269718.000000  269718.000000  ...  210917.000000  208204.000000
mean     1978.626862   68263.967381  ...     175.338953      70.699237
std        29.747764   39026.038321  ...      10.518507      14.348346
min      1896.000000       1.000000  ...     127.000000      25.000000
25%      1960.000000   34654.250000  ...     168.000000      60.000000
50%      1988.000000   68233.000000  ...     175.000000      70.000000
75%      2002.000000  102111.000000  ...     183.000000      79.000000
max      2016.000000  135571.000000  ...     226.000000     214.000000

[8 rows x 5 columns]


# usamos a função describe para fazer uma análise geral dos dados numéricos

In [19]:
dados.describe()

ano      id_atleta  ...         altura           peso
count  269718.000000  269718.000000  ...  210917.000000  208204.000000
mean     1978.626862   68263.967381  ...     175.338953      70.699237
std        29.747764   39026.038321  ...      10.518507      14.348346
min      1896.000000       1.000000  ...     127.000000      25.000000
25%      1960.000000   34654.250000  ...     168.000000      60.000000
50%      1988.000000   68233.000000  ...     175.000000      70.000000
75%      2002.000000  102111.000000  ...     183.000000      79.000000
max      2016.000000  135571.000000  ...     226.000000     214.000000

[8 rows x 5 columns]

In [20]:
dados.select_dtypes(include=['number']).head

<bound method NDFrame.head of          ano  id_atleta  idade  altura  peso
0       1896      35698   22.0     NaN   NaN
1       1896      35698   22.0     NaN   NaN
2       1896      35698   22.0     NaN   NaN
3       1896      35698   22.0     NaN   NaN
4       1896      35698   22.0     NaN   NaN
...      ...        ...    ...     ...   ...
269713  2016     133071   21.0   156.0  50.0
269714  2016     133346   26.0   166.0  50.0
269715  2016     133346   26.0   166.0  50.0
269716  2016     134826   22.0   168.0  54.0
269717  2016     120422   24.0   184.0  72.0

[269718 rows x 5 columns]>

abrimos a base de dados referente a idade

In [22]:
dados['idade']

0         22.0
1         22.0
2         22.0
3         22.0
4         22.0
          ... 
269713    21.0
269714    26.0
269715    26.0
269716    22.0
269717    24.0
Name: idade, Length: 269718, dtype: float64

usamos a função describe para fazer a análise dos dados de idade

In [23]:
dados['idade'].describe()

count    260410.000000
mean         25.454487
std           6.163208
min          10.000000
25%          21.000000
50%          24.000000
75%          28.000000
max          97.000000
Name: idade, dtype: float64

A base tem dados tem pouco mais de 26 mil observações. O que corresponde a 26 mil medalhistas. Como um atleta pode ter ganhado mais de uma medalha, o número de observações é superior ao número de atletas. O atleta mais jovem a ganhar uma medalha tinha 10 anos. O mais velho tinha 97 anos. Apesar dessa diferença, há uma grande concentração de medalhistas em uma faixa etária. Até 25% dos atletas (primeiro quartil) tinha até 21 anos. Metade (mediana) alcançava a idade máxima de 24 anos. 75% dos atletas (terceiro quartil) tinha 28 anos. A idade média ficou em 25 anos. Indo além da análise de dados, uma pesquisa na internet ajuda a explicar as idades máxima e mínima encontradas na tabela. O atleta mais jovem, de 10 anos, foi o grego Dimitrios Loundras, que ganhou a medalha de bronze nas barras paralelas por equipes da ginástica artística, nas Olímpiadas de 1896, em Atenas. Já a idade do atleta mais velho registrada nas reportagens não corresponde à base de dados. O noticiário indica que o medalhista mais velho foi o sueco Oscar Swahn, que ganhou a medalha de prata aos 73 anos, nas Olimpíadas da Antuérpia, em 1920, na prova de tiro ao veado. É provável que exista um erro na base de dados. 

In [ ]:
ouro = dados.loc[dados['medalha'] == 'Gold']
ouro


ano  edicao  ...                                    evento medalha
0       1896  Summer  ...              Athletics Men's 1,500 metres    Gold
1       1896  Summer  ...                Athletics Men's 800 metres    Gold
7       1896  Summer  ...       Swimming Men's 500 metres Freestyle    Gold
10      1896  Summer  ...               Cycling Men's 12-Hours Race    Gold
19      1896  Summer  ...  Weightlifting Men's Unlimited, Two Hands    Gold
...      ...     ...  ...                                       ...     ...
184080  1998  Winter  ...             Ice Hockey Women's Ice Hockey    Gold
184084  1998  Winter  ...             Ice Hockey Women's Ice Hockey    Gold
184099  1998  Winter  ...             Ice Hockey Women's Ice Hockey    Gold
184105  1998  Winter  ...             Ice Hockey Women's Ice Hockey    Gold
184113  1998  Winter  ...             Ice Hockey Women's Ice Hockey    Gold

[9344 rows x 15 columns]

Analisamos também o perfil dos ganhadores das medalhas de ouro, usando a função dados.loc, que permite recortar linhas da tabela. Com a função describe, pudemos analisar os dados. Chegamos à conclusão que a idade média dos vencedores é 25,7 anos. O mais jovem venceu com 13 anos e o mais velho, com 64 anos. Até 25% ganharam com no máximo 22 anos. Metade foi com até 25 anos e 75% com 28 anos. A altura média dos vencedores foi de 1,77 metro. O mais baixo tinha 1,36m e o mais alto, 2,23 metros. Já em relação ao peso, o mais leve tinha 30 quilos, o mais pesado, 170 quilos. Metade dos vencedores pesava até 74 quilos, mesmo valor da média.

In [ ]:
ouro.describe()

ano      id_atleta        idade       altura         peso
count  9344.000000    9344.000000  9197.000000  6519.000000  6267.000000
mean   1958.139127   68676.346426    25.752202   177.774659    74.171533
std      30.146318   38661.476624     6.193437    10.794263    14.682199
min    1896.000000       4.000000    13.000000   136.000000    30.000000
25%    1932.000000   35690.500000    22.000000   170.000000    64.000000
50%    1964.000000   68621.500000    25.000000   178.000000    74.000000
75%    1984.000000  102806.000000    28.000000   185.000000    83.000000
max    1998.000000  135553.000000    64.000000   223.000000   170.000000